specs/core/1_custody-game.md

```diff
@@ -688,6 +682,17 @@ Append this to `process_final_updates(state)`:
 ```python
     # Clean up exposed RANDAO key reveals
     state.exposed_derived_secrets[current_epoch % EARLY_DERIVED_SECRET_PENALTY_MAX_FUTURE_EPOCHS] = []
+    # Reset withdrawable epochs if challenge records are empty
+    records = state.custody_chunk_challenge_records + state.bit_challenge_records
+    validator_indices_in_records = set(
+        [record.challenger_index for record in records] + [record.responder_index for record in records]
+    )
+    for index, validator in enumerate(state.validator_registry):
+        if index not in validator_indices_in_records:
+            if validator.exit_epoch == FAR_FUTURE_EPOCH:
+                validator.withdrawable_epoch = FAR_FUTURE_EPOCH
```

Comment:
What's the case of that `validator.exit_epoch == FAR_FUTURE_EPOCH && validator.withdrawable_epoch != FAR_FUTURE_EPOCH` happens?

Reply:
There is no place in the current code where this can happen.

`withdrawable_epoch` is only changed to `!= FAR_FUTURE_EPOCH` in `slash_validator` and `initiate_validator_exit`. Both of these instances also adjust `exit_epoch` to be `!= FAR_FUTURE_EPOCH`
The only place where `exit_epoch` is set is in [`initiate_validator_exit`](https://github.com/ethereum/eth2.0-specs/blob/dev/specs/core/0_beacon-chain.md#initiate_validator_exit) and both `exit_epoch` and `withdrawable_epoch` are changed at the same time.

Reply:
@djrtwo I feel the same way. :)

This if-else statement can be simplified to:

```python
if validator.exit_epoch != FAR_FUTURE_EPOCH:
    validator.withdrawable_epoch = validator.exit_epoch + MIN_VALIDATOR_WITHDRAWABILITY_DELAY
```

But, maybe worth to keep it due to precaution? /cc @vbuterin

Reply:
I think you could actually do the following to very specifically fix the case we care about.

```python
if validator.exit_epoch != FAR_FUTURE_EPOCH and validator.withdrawable_epoch == FAR_FUTURE_EPOCH:
    validator.withdrawable_epoch = validator.exit_epoch + MIN_VALIDATOR_WITHDRAWABILITY_DELAY
```